### PR TITLE
Access Atari OS variables by structure.

### PR DESCRIPTION
**Pull request by IrgendwerA8**
@groessler : You may remember my email from Friday, January 28, 2011 10:18 PM...
Only eight years later I'm able to deliver...

### CODE REVIEW

File: include/_atarios.h [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+#define IOCB_MKDIR       0x2A  /* make directory (MyDOS/SpartaDOS) */
+#define IOCB_RMDIR       0x2B  /* remove directory (SpartaDOS) */
+#define IOCB_CHDIR_SPDOS 0x2C  /* change directory (SpartaDOS) */
+#define IOCB_GETCWD      0x30  /* get current directory (MyDOS/SpartaDOS) */
+#define IOCB_FORMAT      0xFE  /* format */
+
+
+/* I/O control block */
+
+typedef struct {
```
Comment (groessler) [inline include/_atarios.h]: I personally don't like these kinds of "typedef"s. If one is not directly working on the program it's always more work to find out if a typedef is an int var, a pointer, a struct, etc.

To me a `struct __iocb blah;` is more readable than a `IOCB blah;` in the program's source code.

Please name these structs like they were named before, so
```
struct __iocb {
  ...
};
```

If you want typedefs, define them afterwards, e.g.
`typedef struct __iocb IOCB;`

**Comment by groessler** [comment]
> You may remember my email from Friday, January 28, 2011 10:18 PM...

Hmm. I've looked at my archive but haven't found it. Are you sure that the date is correct?

> Only eight years later I'm able to deliver...

No problem. My stance is "better late than never" :-)

Please check: your PR doesn't build.

regards,
chris

File: include/atari.h [hunk 2]
```diff
@@ -358,6 +356,10 @@ extern void atrx15p2_tgi[];
 /* Define hardware and where they're mapped in memory                        */
 /*****************************************************************************/
 
+#include <_atarios.h>
+#define OS (*(struct __os*)0x0000)
+#define BASIC (*(struct __basic)0x0080)
```
Comment (greg-king5) [inline include/atari.h]: A `*` is missing.

**Comment by IrgendwerA8** [comment]
> Hmm. I've looked at my archive but haven't found it. Are you sure that the date is correct?

The answer I received from you is from 06-Feb-2011:

"bitte entschuldige, ich hatte die letzte Zeit Einiges um die Ohren und habe Dich total vergessen."
:)

>Please check: your PR doesn't build.

Noticed that this morning - sorry. Fixed.

File: include/atari.h (hunk 2 above)
Comment (IrgendwerA8) [reply include/atari.h]: Fixed. You get the star for the review!

File: include/_atarios.h (hunk 1 above)
Comment (IrgendwerA8) [reply include/_atarios.h]: Added/changed.

**Comment by oliverschmidt** [comment]
@groessler / @greg-king5 - Please merge when you see fit.

File: include/atari.h [hunk 3]
```diff
@@ -489,129 +491,5 @@ extern void atrx15p2_tgi[];
...
-    unsigned char command;    /* command */
-    unsigned char status;     /* command type / status return */
-    void          *buffer;    /* pointer to buffer */
-    unsigned char timeout;    /* device timeout in seconds */
-    unsigned char unused;
-    unsigned int  xfersize;   /* # of bytes to transfer */
-    unsigned char aux1;       /* 1st command auxiliary byte */
-    unsigned char aux2;       /* 2nd command auxiliary byte */
-};
-#define DCB (*(struct __dcb *)0x300)
```
Comment (groessler) [inline include/atari.h]: What happened to this structure in the new version? It's somehow gone....
Please fix.

Comment (greg-king5) [reply include/atari.h]: [These lines](https://github.com/cc65/cc65/blob/2362db22f7bedf79fa0489f2e018e2431a890144/include/_atarios.h#L542-L604) claim that the DCB is 64 bytes long!  But, `OS.timer1` doesn't look as though it belongs to the Device Control Block.

Comment (IrgendwerA8) [reply include/atari.h]: Just created a new PR which fixes both issues. Thanks for the review!

Comment (groessler) [reply include/atari.h]: AFAIK the DCB is 0x0C bytes long.

**Pull request merged**